pkg/controller/cache/cachesubnetgroup/controller.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	if commonaws.BoolValue(p.Spec.UseServiceAccount) {
+		awsClient, err := c.newClientFn(ctx, []byte{}, p.Spec.Region, awsclients.UsePodServiceAccount)
+		return &external{client: awsClient}, errors.Wrap(err, errNewClient)
+	}
+
+	if p.GetCredentialsSecretReference() == nil {
+		return nil, errors.New(errGetProviderSecret)
+	}
+
+	s := &corev1.Secret{}
+	n := types.NamespacedName{Namespace: p.Spec.CredentialsSecretRef.Namespace, Name: p.Spec.CredentialsSecretRef.Name}
+	if err := c.client.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+	awsClient, err := c.newClientFn(ctx, s.Data[p.Spec.CredentialsSecretRef.Key], p.Spec.Region, awsclients.UseProviderSecret)
+	return &external{client: awsClient}, errors.Wrap(err, errNewClient)
+}
+
+type external struct {
+	client elasticache.Client
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.CacheSubnetGroup)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotSubnetGroup)
+	}
+
+	resp, err := e.client.DescribeCacheSubnetGroupsRequest(&awscache.DescribeCacheSubnetGroupsInput{
+		CacheSubnetGroupName: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	if err != nil || resp.CacheSubnetGroups == nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(elasticache.IsSubnetGroupNotFound, err), errDescribeSubnetGroup)
+	}
+
+	sg := resp.CacheSubnetGroups[0]
+
+	cr.Status.AtProvider = v1alpha1.CacheSubnetGroupExternalStatus{
+		VpcID: aws.StringValue(sg.VpcId),
+	}
+
+	isUpToDate := elasticache.IsSubnetGroupUpToDate(cr.Spec.ForProvider, sg)
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: isUpToDate,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.CacheSubnetGroup)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotSubnetGroup)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	_, err := e.client.CreateCacheSubnetGroupRequest(&awscache.CreateCacheSubnetGroupInput{
+		CacheSubnetGroupDescription: aws.String(cr.Spec.ForProvider.Description),
+		CacheSubnetGroupName:        aws.String(meta.GetExternalName(cr)),
+		SubnetIds:                   cr.Spec.ForProvider.SubnetIds,
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(resource.Ignore(elasticache.IsAlreadyExists, err), errCreateSubnetGroup)
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.CacheSubnetGroup)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotSubnetGroup)
+	}
+
+	// If the list of spec.subnetIds is different from the resource subnetIds,
+	// this call will return non-nil error
```

Comment:
Does this mean we can effectively only update the description field? It seems odd that they allow the caller to supply a list of subnet IDs in the update call, but then return an error if they differ from the current list of subnet IDs.

Reply:
@negz  I think the comment was misleading. I was able to update a CacheSubnetGroup with new `subnetIds`.